storage/range.go

```diff
@@ -235,13 +235,35 @@ func (r *Range) String() string {
 	return fmt.Sprintf("range=%d (%s-%s)", r.Desc().RaftID, r.Desc().StartKey, r.Desc().EndKey)
 }
 
-// start begins by gossiping the cluster ID (only happens if the range
-// is the first in the key space). This is necessary to get the cluster
-// started from scratch.
+// start begins by gossiping the configs. It also gossips the sentinel if we're
+// the first range (the store will do this regularly, but this is a good time
+// to get started quickly).
 func (r *Range) start() {
-	if r.IsFirstRange() {
-		r.startGossip(r.rm.Stopper())
+	// Being the first range comes with extra responsibility for the gossip
+	// sentinel and first range metadata.
+	// TODO(tschottdorf) really want something more streamlined, such as:
+	//
+	// if r.rm.Stopper().StartTask() {
+	// 	go func() {
+	// 		r.maybeGossipFirstRangeWithLease()
+	// 		r.rm.Stopper().FinishTask()
+	// 	}()
+	// }
+	//
+	// but
+	// a) in some tests that goroutine gets stuck acquiring the lease, probably
+	//    due to Raft losing the command
+	// b) other times, the lease fails because the MultiRaft group doesn't exist
+	//    yet
+	// b) it complicates testing because there's little way of knowing who
+	//    will get the lease first, and many tests are too low level to do
+	//    the appropriate retries.
+	//
+	// Instead we do this for now:
+	if desc := r.Desc(); desc != nil && len(desc.Replicas) == 1 && r.IsFirstRange() {
```

Comment:
Why so insistent on the length of the replicas being 1? This feels very test-driven. How about if we manually call Range.gossipFirstRange() in the appropriate test contexts?